### PR TITLE
fix 13043

### PR DESCRIPTION
**Pull request by dsyme**
@KevinRansom This looks like the right fix for https://github.com/dotnet/fsharp/issues/13043 (instead of https://github.com/dotnet/fsharp/pull/13064). Could you integrate with your testing please? Thanks

I'd still like to understand the NRE when the debug point wasn't being skipped.  I think I know why but will look into it more

### CODE REVIEW

**Comment by vzarytovskii** [comment]
It seems there's still NRE happening.

File: src/fsharp/TypedTreeOps.fs [hunk 1]
```diff
@@ -5142,34 +5142,40 @@ let freeInModuleOrNamespace opts mdef =
 //---------------------------------------------------------------------------
 
 let rec stripLambda (expr, ty) = 
-    match expr with 
+    match stripDebugPoints expr with
```
Comment (vzarytovskii) [inline src/fsharp/TypedTreeOps.fs:5145]: These will run only with `--optimize+`, right?

Comment (KevinRansom) [reply src/fsharp/TypedTreeOps.fs:5145]: Nice ... I saw this last night, which is why the il compare test is commented out, but it was getting late and I needed my beauty sleep.

**Comment by dsyme** [comment]
> Although, it fails if I turn off optimizations in both FSI and FSC:

OK, thanks, there must be another place we need to fix

**Comment by dsyme** [comment]
@KevinRansom I merged with main and pushed additional `stripDebugPoints` where we are matching on a `Expr.Lambda` suring codegen.

**Comment by dsyme** [comment]
@KevinRansom Thanks - this is ready

**Comment by dsyme** [comment]
@KevinRansom @vzarytovskii Don't forget to apply this to the appropriate branch after it is merged, I think this needs to be done manually prior to the insertion, correct?

thanks

**Comment by KevinRansom** [comment]
@dsyme , it's really too late for 17.2 and probably doesn't meet the bar for this late in the game.

**Comment by dsyme** [comment]
> @dsyme , it's really too late for 17.2 and probably doesn't meet the bar for this late in the game.

Huh? We specifically agreed a deadline of Wednesday.  It's Monday.  And it surely meets the bar.

**Comment by KevinRansom** [comment]
Okay, I have just chatted with @dsyme, we are going to trim the fix, to just the essentials.  And take the fix to war room.

**Review by KevinRansom — APPROVED** [review]

**Pull request merged**